tests/proxy_unit_tests/test_proxy_config_unit_test.py

```diff
@@ -231,6 +233,45 @@ def test_add_callbacks_from_db_config():
     litellm._known_custom_logger_compatible_callbacks = []
 
 
+def test_s3_config_temp_file_cleanup():
```

Comment:
please add a mocked test in test_litellm/ this test looks like it wont run on CI/CD. It's better to add a test that actually tests this change so we can prevent future regressions.